src/longwave_radiation.jl

```diff
@@ -0,0 +1,203 @@
+        fband[jtemp, 1] = eps1 - (fband[jtemp, 2] + fband[jtemp, 3] + fband[jtemp, 4])
+    end
+    for jb = 1:4
+        for jtemp = 100:199
+            fband[jtemp, jb] = fband[200, jb]
+        end
+        for jtemp = 321:400
+            fband[jtemp, jb] = fband[320, jb]
+        end
+    end
+    return fband
+end
+
+
+"""
+    function radlw_down!(
+        column::ColumnVariables{NF}, model::PrimitiveEquationModel
+    ) where {NF<:AbstractFloat}
+
+Compute the downward flux of longwave radiation. Inputs variables are `temp``, `wvi`, `tau2`.
+Output column varables are `fsfcd`, `dfabs`, `flux`, `st4a`.
+"""
+function radlw_down!(
+    column::ColumnVariables{NF}, model::PrimitiveEquationModel
+) where {NF<:AbstractFloat}
+
+    @unpack nlev, temp, wvi, tau2 = column
+    @unpack nband, epslw, emisfc, sbc, fband, n_stratosphere_levels = model.parameters
+    
+    # 1. Blackbody emission from atmospheric levels.
+    #    The linearized gradient of the blakbody emission is computed
+    #    from temperatures at layer boundaries, which are interpolated 
+    #    assuming a linear dependence of T on log_sigma.
+    #    Above the first (top) level, the atmosphere is assumed isothermal.
+    # 
+    # Temperature at level boundaries
+    for k=1:nlev-1
+        column.st4a[k, 1] = temp[k] + wvi[k, 2] * (temp[k + 1] - temp[k])
+    end
+    
+    # Mean temperature in stratospheric layers
+    for k=1:n_stratosphere_levels-1
+        column.st4a[k, 2] = NF(0.75) * temp[k] + NF(0.25) * column.st4a[k, 1]
+        column.st4a[k+1, 2] = NF(0.50) * temp[k+1] + NF(0.25) * (column.st4a[k, 1] + column.st4a[k+1, 1])
+    end
+
+    # Temperature gradient in tropospheric layers
+    anish  = NF(0.5 * 1.)
+    for k = n_stratosphere_levels+1:nlev-1
+        column.st4a[k, 2] = anish * max(column.st4a[k, 1] - column.st4a[k - 1, 1], NF(0.))
+    end
+    column.st4a[nlev, 2] = anish * max(temp[nlev] - column.st4a[nlev-1, 1], NF(0.))
+
+    # Blackbody emission in the stratosphere
+    for k = 1:n_stratosphere_levels
+        column.st4a[k, 1] = sbc * column.st4a[k, 2]^4
+        column.st4a[k, 2] = NF(0.)
+    end
+
+    # Blackbody emission in the troposphere
+    for k = n_stratosphere_levels+1:nlev
+            st3a = sbc * temp[k]^3
+            column.st4a[k, 1] = st3a * temp[k]
+            column.st4a[k, 2] = NF(4.) * st3a * column.st4a[k, 2]
+    end
+
+    # 2.0 Initialization of fluxes
+    column.fsfcd = 0.
+    column.dfabs .= 0.
+
+    # 3.0 Emission ad absorption of longwave downward flux.
+    #    For downward emission, a correction term depending on the
+    #    local temperature gradient and on the layer transmissivity is
+    #    added to the average (full - level) emission of each layer.
+
+    # 3.1  Stratosphere
+    k = n_stratosphere_levels-1
+    for jb = 1:nband-2
+        emis = 1. - tau2[k, jb]
+        brad = fband[convert(Int, temp[k]), jb] * (column.st4a[k, 1] + emis * column.st4a[k, 2])
```

Comment:
Okay now I see how `fband` is supposed to be used. Interesting! So on the computational side we have currently 1 conversion to integer then 1 pull from memory. Alternatively we can at some point think about recomputing the mahts behind fband on the fly: Something like `(NF(0.148) - NF(3.0e-6) * (jtemp - 247)^2) * eps1` looks like 4 arithmetic operations to me, which can even be fused.